mlpp_lib/standardizers.py

```diff
@@ -16,59 +17,64 @@ class Standardizer:
     Standardizes data in a xarray.Dataset object.
     """
 
-    mean: dict[str, float] = field(init=False)
-    std: dict[str, float] = field(init=False)
-    fillvalue: dict[str, float] = field(init=False)
+    mean: xr.Dataset = field(default=None)
+    std: xr.Dataset = field(default=None)
+    fillvalue: dict[str, float] = field(default=-5)
 
-    def fit(self, dataset: xr.Dataset, dims: list = None):
-        LOGGER.info("Calculating mean and standard deviation")
-        self.mean = dataset.mean(dims).persist()
-        self.std = dataset.std(dims).persist()
+    def fit(self, dataset: xr.Dataset, dims: Optional[list] = None):
+
+        self.mean = dataset.mean(dims).compute().copy()
+        self.std = dataset.std(dims).compute().copy()
         self.fillvalue = -5
         # Check for near-zero standard deviations and set them equal to one
         self.std = xr.where(self.std < 1e-6, 1, self.std)
 
-    def transform(self, dataset: xr.Dataset) -> xr.Dataset:
+    def transform(self, *datasets: xr.Dataset) -> tuple[xr.Dataset, ...]:
         if self.mean is None:
             raise ValueError("Standardizer wasn't fit to data")
-        for var in dataset.data_vars:
-            assert var in self.mean.data_vars, f"{var} not in Standardizer"
-        return (
-            ((dataset - self.mean) / self.std).astype("float32").fillna(self.fillvalue)
-        )
 
-    def inverse_transform(self, dataset: xr.Dataset) -> xr.Dataset:
+        def f(ds: xr.Dataset):
+            for var in ds.data_vars:
+                assert var in self.mean.data_vars, f"{var} not in Standardizer"
+            return (
+                ((ds - self.mean) / self.std).astype("float32").fillna(self.fillvalue)
+            )
+
+        return tuple(f(ds) for ds in datasets)
+
+    def inverse_transform(self, *datasets: xr.Dataset) -> xr.Dataset:
         if self.mean is None:
             raise ValueError("Standardizer wasn't fit to data")
-        for var in dataset.data_vars:
-            assert var in self.mean.data_vars, f"{var} not in Standardizer"
-        dataset = xr.where(dataset > self.fillvalue, dataset, np.nan)
-        return (dataset * self.std + self.mean).astype("float32")
 
-    def to_dict(self) -> dict:
```

Comment:
the `to_dict()` method is used after training to persist the standardizer to mlflow, so we still need it somehow

Reply:
I'll implement it, good catch